### PR TITLE
Do not close new campaign window when starting a campaign results in an error, fixes #4810

### LINKED ISSUE #4810: [Idea] The game should not close out of the ad/marketing voucher when the player lacks funds
Okay let's say I want to start a new marketing campaign and go to set how many weeks it should last as seen below. 
bashful beach 2016-11-15 16-44-53
If you notice I lack the funds, if I hit okay the window just disappears with an error saying I lack funds. This gets annoying after a while since it keeps you from making changes to the marketing campaign, by keeping the window open and displaying the error the player can then make the changes needed (like lowing the amount of weeks) without having to reopen the voucher/campaign window. Makes no sense for this to be the only window that closes on an error.
Here's a screenshot of what happens when I hit "start this campaign" button:

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

**Comment by IntelOrca** [comment]
This is good, although it still won't work if the master server declines the request in multiplayer.

File: src/openrct2/management/marketing.c [hunk 1]
```diff
@@ -138,10 +138,16 @@ void marketing_set_guest_campaign(rct_peep *peep, sint32 campaign)
 	}
 }
 
-void marketing_start_campaign(sint32 type, sint32 rideOrItem, sint32 numWeeks)
+bool marketing_start_campaign(sint32 type, sint32 rideOrItem, sint32 numWeeks)
 {
 	gGameCommandErrorTitle = STR_CANT_START_MARKETING_CAMPAIGN;
-	game_do_command(0, (numWeeks << 8) | GAME_COMMAND_FLAG_APPLY, 0, (rideOrItem << 8) | type, GAME_COMMAND_START_MARKETING_CAMPAIGN, 0, 0);
+	sint32 retVal = game_do_command(0, (numWeeks << 8) | GAME_COMMAND_FLAG_APPLY, 0, (rideOrItem << 8) | type, GAME_COMMAND_START_MARKETING_CAMPAIGN, 0, 0);
+	if (retVal == MONEY32_UNDEFINED) {
```
Comment (AaronVanGeffen) [inline src/openrct2/management/marketing.c]: So, `return retVal != MONEY32_UNDEFINED;`?

**Comment by Gymnasiast** [comment]
@IntelOrca Hm, what would it get back if the master server declines it?

File: src/openrct2/management/marketing.c (hunk 1 above)
Comment (Gymnasiast) [inline src/openrct2/management/marketing.c]: Yes. I find this easier to read, though.

**Comment by IntelOrca** [comment]
@Gymnasiast you have to look at how create ride and other commands which use the response work. The new game command infrastructure will simplify it as it will allow a callback function to be used as an asynchronous reply.

**Comment by Gymnasiast** [comment]
@IntelOrca Is it okay if we leave that for when the new structure is actually in place?

**Comment by IntelOrca** [comment]
@Gymnasiast done it for you, its not perfect as it may close the wrong campaign window if the RTT is long in multiplayer.

**Comment by Gymnasiast** [comment]
@IntelOrca Thanks!

**Pull request merged**